pre-commit-check.sh

```diff
@@ -7,13 +7,16 @@
 set -e
 
 # Check formatting
-cargo +nightly fmt --all
+cargo +nightly fmt --check --all
 
 # Check clippy
-cargo clippy --all-features --all-targets -- -D warnings
+cargo clippy --all-features --all-targets --locked -- -D warnings
 
 # Build all targets
-# cargo build --all-targets
+cargo build --all-targets --locked
 
 # Run tests
-# cargo test
+cargo test --locked
```

Comment:
not sure why these were commented out, but find it useful to run this before pushing to GH

Reply:
@yuroitaki this was only to make it faster. The ci runs the tests anyway.
So both approaches are fine